### PR TITLE
[4.2] user profile dob field

### PR DESCRIPTION
**Pull request by brianteeman**
To test enable the user profile plugin and update the date of birth field for your user account

Pull Request for Issue #28883

This is a simpler alternative to #38057 that does not use a customised field

image
 .

### CODE REVIEW

**Comment by AndySDH** [comment]
Nice! 

This file should also be deleted: layouts/plugins/user/profile/fields/dob.php

Other than that, it looks great!

**Comment by AndySDH** [comment]
Nice! Should the whole profile/fields folder be deleted since there's no file in there anymore?

**Comment by brianteeman** [comment]
it is but you can tell from the way github displays the diff

image

**Comment by AndySDH** [comment]
Did you mark the parent /layouts/plugins/user/ for deletion in the script (line 7844)? That one should not be deleted as it also contains 'terms' and 'token' subfolders

**Comment by brianteeman** [comment]
no

**Comment by richard67** [comment]
> Pull Request for #2883

@brianteeman It seems there is a digit missing in the referred issue number.

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -6474,6 +6474,9 @@ public function deleteUnexistingFiles($dryRun = false, $suppressOutput = false)
 			'/plugins/twofactorauth/yubikey/tmpl/form.php',
 			'/plugins/twofactorauth/yubikey/yubikey.php',
 			'/plugins/twofactorauth/yubikey/yubikey.xml',
+			// From 4.2
+			'/plugins/user/profile/src/Field/DobField.php',
+			'/layouts/plugins/user/profile/fields/dob.php',
```
Comment (richard67) [inline administrator/components/com_admin/script.php]: The files should be alpha ordered and the section title should be "// From 4.2.0-beta1 to 4.2.0-beta2". It would be maybe better if you would revert the changes in script.php and leave that to me.

File: administrator/components/com_admin/script.php [hunk 2]
```diff
@@ -7835,6 +7838,10 @@ public function deleteUnexistingFiles($dryRun = false, $suppressOutput = false)
 			'/plugins/twofactorauth/totp',
 			'/plugins/twofactorauth',
 			'/libraries/vendor/nyholm/psr7/doc',
+			// From 4.2
```
Comment (richard67) [inline administrator/components/com_admin/script.php]: The section title should be "// From 4.2.0-beta1 to 4.2.0-beta2". It would be maybe better if you would revert the changes in script.php and leave that to me.

File: administrator/components/com_admin/script.php (hunk 1 above)
Comment (richard67) [reply administrator/components/com_admin/script.php]: See #38039 . I would add all necessary changes to that PR after the corresponding PRs have been merged, and will make my PR ready before the next beta.

**Comment by brianteeman** [comment]
> > Pull Request for #2883
> 
> @brianteeman It seems there is a digit missing in the referred issue number.

it was missing an 8 - i have updated the reference

**Comment by Kostelano** [comment]
Why a description that duplicates a hint?

**Comment by brianteeman** [comment]
> Why a description that duplicates a hint?

If there was only a hint then you wouldnt know the date format when the user profile is displayed as you obviously only see the hint when in edit mode.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 40a101325d00ca4e2bb9c963e93560d6656c1756<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38065">issues.joomla.org/tracker/joomla-cms/38065</a>.</sub>

**Comment by AndySDH** [comment]
I have tested this item :white_check_mark: successfully on 40a101325d00ca4e2bb9c963e93560d6656c1756<br /><br />Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38065">issues.joomla.org/tracker/joomla-cms/38065</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38065">issues.joomla.org/tracker/joomla-cms/38065</a>.</sub>

**Comment by richard67** [comment]
Thanks
